Anature/src/application/Startup.java

```diff
@@ -348,4 +375,171 @@ public static String getPlayerName()
 	{
 		return mPlayer.getName();
 	}
+
+	/*
+	 * SAVING FUNCTIONALITY
+	 */
+
+	private interface SavableItem
+	{
+		public Object getItem();
+
+		public void setItem(Object object);
+	}
+
+	private enum SaveItem implements SavableItem
+	{
+		Player
+		{
+			public Object getItem()
+			{
+				return mPlayer;
+			}
+
+			public void setItem(Object object)
+			{
+				mPlayer = (Player) object;
+			}
+		},
+		StarterTownModel
+		{
+			public Object getItem()
+			{
+				return mStarterTownModel;
+			}
+
+			public void setItem(Object object)
+			{
+				mStarterTownModel = (StarterTownModel) object;
+			}
+		},
+		PathOneModel
+		{
+			public Object getItem()
+			{
+				return mPathOneModel;
+			}
+
+			public void setItem(Object object)
+			{
+				mPathOneModel = (PathOneModel) object;
+			}
+		},
+		CurrentSceneType
+		{
+			public Object getItem()
+			{
+				return mCurrSceneType;
+			}
+
+			public void setItem(Object object)
+			{
+				mCurrSceneType = (SceneType) object;
+				changeScene(mCurrSceneType, null);
+			}
+		},
+		CurrentPlayerDirection
+		{
+			public Object getItem()
+			{
+				return mCurrentCell.getPlayerFacing();
+			}
+
+			public void setItem(Object object)
+			{
+				mCurrentCell.setPlayerFacing((Direction) object);
+			}
+		},
+		CurrentPlayerXCoordinate
+		{
+			public Object getItem()
+			{
+				return mCurrentCell.getPlayer()
+						.getX();
+			}
+
+			public void setItem(Object object)
+			{
+				mCurrentCell.getPlayer()
+						.setX((double) object);
+			}
+		},
+		CurrentPlayerYCoordinate
+		{
+			public Object getItem()
+			{
+				return mCurrentCell.getPlayer()
+						.getY();
+			}
+
+			public void setItem(Object object)
+			{
+				mCurrentCell.getPlayer()
+						.setY((double) object);
+			}
+		}
+	}
+
+	public static boolean save()
+	{
+
+		ArrayList<Object> itemsToSave = new ArrayList<Object>();
+
+		for(SaveItem item : SaveItem.values())
+		{
+			itemsToSave.add(item.getItem());
+		}
+
+		try
+		{
+			FileOutputStream fileOutputStream = new FileOutputStream(new File("./save.save"));
+			ObjectOutputStream out = new ObjectOutputStream(fileOutputStream);
+
+			out.writeObject(itemsToSave);
+			out.close();
+			fileOutputStream.close();
+		}
+
+		catch(IOException e)
+		{
+			e.printStackTrace();
+		}
+
+		mCurrentController.saveLoadUpdates();
+
+		return true;
+	}
+
+	@SuppressWarnings("unchecked")
+	public static boolean load()
+	{
+		ArrayList<Object> objRead = new ArrayList<Object>();
+		FileInputStream fileInStream;
+		ObjectInputStream in;
+
```

Comment:
You also have to do mCurrentController.saveLoadUpdates() before the load.
This is because you have to set the CanTalkToTrainers to true in the current controller and the new controller.